cmd/cli/version/version.go

```diff
@@ -91,26 +98,95 @@ var serverVersionColumn = output.TableColumn[Versions]{
 	Value:        func(v Versions) string { return v.ServerVersion.GitVersion },
 }
 
-// Run executes version command
+var updateMessageColumn = output.TableColumn[Versions]{
+	ColumnConfig: table.ColumnConfig{Name: "Update Message"},
+	Value:        func(v Versions) string { return v.UpdateMessage },
+}
+
 func (oV *VersionOptions) Run(ctx context.Context, cmd *cobra.Command) error {
-	var (
-		versions Versions
-		columns  []output.TableColumn[Versions]
-	)
-
-	versions.ClientVersion = version.Get()
-	columns = append(columns, clientVersionColumn)
-
-	if !oV.ClientOnly {
-		serverVersion, err := util.GetAPIClient(ctx).Version(ctx)
-		if err != nil {
-			log.Ctx(ctx).Error().Err(err).Msgf("could not get server version")
-			return err
-		}
-
-		versions.ServerVersion = serverVersion
-		columns = append(columns, serverVersionColumn)
+    var (
+        versions Versions
+        columns  []output.TableColumn[Versions]
+    )
+
+    versions.ClientVersion = version.Get()
+    columns = append(columns, clientVersionColumn)
+
+    if !oV.ClientOnly {
+        serverVersion, err := util.GetAPIClient(ctx).Version(ctx)
+        if err != nil {
+            return fmt.Errorf("error running version: %w", err)
+        }
+
+        versions.ServerVersion = serverVersion
+        columns = append(columns, serverVersionColumn)
+    }
+
+    UserIDStr, err := config.GetClientID()
+    if err != nil {
+        return fmt.Errorf("error getting UserID: %w", err)
+    }
+
+    versions.OperatingSystem = runtime.GOOS
+    versions.Architecture = runtime.GOARCH
+    versions.UserID = UserIDStr
+
+    updateMessage := checkForUpdates(ctx, versions.ClientVersion.GitVersion, "", versions.OperatingSystem, versions.Architecture, versions.UserID)
+
+    // Print the update message only if --output flag is not used
+    if oV.OutputOpts.Format == output.TableFormat {
+        fmt.Println(updateMessage)
+    } else {
+        versions.UpdateMessage = updateMessage
+        columns = append(columns, updateMessageColumn)
+    }
+
+    return output.OutputOne(cmd, columns, oV.OutputOpts, versions)
+}
+
+
+type ServerResponse struct {
+	Version string `json:"version"`
+	Message string `json:"message"`
+}
+
+func checkForUpdates(ctx context.Context, currentClientVersion, currentServerVersion, os, arch, userID string) string {
```

Comment:
If you just pass in a `model.BuildVersionInfo` instead of `currentClientVersion, os, arch` you will still get all of the fields you need:
```suggestion
func checkForUpdates(ctx context.Context, currentClientVersion, currentServerVersion *model.BuildVersionInfo, userID string) string {
    os := currentClientVersion.GOOS
    arch := currentClientVersion.GOARCH
```